src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -1905,6 +1912,41 @@ public void testOnTheFlyTabixCreation() throws IOException {
         }
     }
 
+    @Test
+    public void testCreateVcfWriterOnNio() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path outputGZIP = jimfs.getPath("testOnTheFlyTabixCreation.vcf.gz");
+            final Path tabixIndex = outputGZIP.resolveSibling(outputGZIP.getFileName().toString() + TabixUtils.STANDARD_INDEX_EXTENSION);
+
+            final File inputGZIPFile = new File(
+                publicTestDir + "org/broadinstitute/hellbender/engine/8_mutect2_sorted.vcf.gz");
+
+            long recordCount = 0;
+            try (final VariantContextWriter vcfWriter = GATKVariantContextUtils.createVCFWriter(
+                outputGZIP,
+                null,
+                false,
+                Options.INDEX_ON_THE_FLY);
+                final FeatureReader<VariantContext> inputFileReader =
+                    AbstractFeatureReader
+                        .getFeatureReader(inputGZIPFile.getAbsolutePath(), new VCFCodec(), false)) {
```

Comment:
Would be simpler to use VCFFileReader(path, false) here.

Reply:
`VCFFileReader` doesn't have the type `FeatureReader` so I can't place it there directly. Could you please expand on how you'd like me to change this code? Also, should I make the same change in `testOnTheFlyTabixCreation` (where it's copied from)?

Reply:
Since I wanted to make sure we tested the actual index, I tried using "suggested change" here for the whole test method. Not sure it did what I thought, but the content is right.

Reply:
Sounds great, thank you! I put that in.

Reply:
@jean-philippe-martin I've never used the "suggest change" feature before, but it doesn't look like this change is in the branch (my suggested change should replace the entire method). You might want to just manually replace the test method with my changes.

Reply:
@cmnbroad 

I wasn't able to merge your suggestion because there were conflicts. Also, I wanted a mix of the two actually.
As you can see in the files here, I added code to query the index that was written and check the number of records. So we now test the actual index, which is what you wanted.

I'd be happy to make more changes if you have specific suggestions, but from here it looks like I have updated the code to address what you wanted and the ball is in your camp.